torch/_meta_registrations.py

```diff
@@ -307,10 +307,17 @@ def calc_conv_nd_return_shape(
                 )
         return ret_shape
 
-    def pick_memory_format():
-        if input_tensor.is_contiguous(memory_format=torch.channels_last):
-            return torch.channels_last
-        elif input_tensor.is_contiguous(memory_format=torch.contiguous_format):
+    def is_channels_last(ten):
+        return ten.is_contiguous(memory_format=torch.channels_last)
```

Comment:
This is too permissive, e.g. weight tensors with 1x1 kernel will be channels-last contiguous, but they won't be channels-last for the purposes of memory format propagation